src/Illuminate/Cache/RedisStore.php

```diff
@@ -272,6 +272,8 @@ protected function serialize($value)
      */
     protected function unserialize($value)
     {
-        return is_numeric($value) ? $value : unserialize($value);
+        $unserialized = @unserialize($value);
```

Comment:
```php
$unserialized = @unserialize($value);

return ($unserialized === false) ? $value : $unserialized;
```
Sorry, I was mistaken, this is why an additional line was added.
It will be compatible with older version of serialized data store, however it will still break when older version coexists or new version degrade to older version. 
@taylorotwell , sorry for the trouble. If you do have plan for this please inform me for a new PR.